### PR TITLE
news: Nix development environment and package

### PR DESCRIPTION
**Pull request by imincik**
Add "Nix development environment and package" news. 

Merge after https://github.com/OSGeo/grass/pull/3990 .

/cc @veroandreo

### CODE REVIEW

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+New option for creating a GRASS development environment and unique way
+of running GRASS directly from a Git source code was implemented using
+[Nix package manager](https://nixos.org/). This idea was presented by
+[Ivan Mincik](https://github.com/imincik)
+during a
+[GRASS GIS Community Meeting](https://grasswiki.osgeo.org/wiki/GRASS_Community_Meeting_Prague_2024)
+in Prague.
+
+Nix development environment provides a stable and reproducible environment
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:18]:
```suggestion
The Nix development environment provides a stable and reproducible environment
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+for all developers and can significantly simplify onboarding process of new
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:19]:
```suggestion
for all developers and can significantly simplify the onboarding process of new
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+contributors.
+
+To start hacking on GRASS, run `nix develop` command in source code root
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:22]:
```suggestion
To start hacking on GRASS, run `nix develop` command within the source code root
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+directory to create a temporary shell environment containing all build
+dependencies with instructions how to build and test the code.
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:24]:
```suggestion
dependencies and get instructions on how to build and test the code.
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ 1.  export PYTHONPATH=$(app/bin/grass --config python_path):$PYTHONPATH
+ 2.  export LD_LIBRARY_PATH=$(app/bin/grass --config path)/lib:$LD_LIBRARY_PATH
+ 3.  pytest
+
+Note: run 'nix flake update' from time to time to update dependencies.
+
+Run 'dev-help' to see this message again. 
+```  
+
+On the other hand, possibility to run GRASS directly from a Git
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:47]:
```suggestion
On the other hand, the possibility to run GRASS directly from the Git
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+source code improves feedback loop during development and will allow to install
+GRASS on any Linux distribution immediately when new version is out.
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:49]:
```suggestion
source code improves the feedback loop during development and will allow to install
GRASS on any Linux distribution immediately after a new version is out.
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+To run latest development version of GRASS directly from a Git source code, run
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:51]:
```suggestion
To run the latest development version of GRASS directly from the Git source code, run
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+`nix run github:OSGeo/grass#grass` command which will compile (if needed) and
+run GRASS in one step.
+
+<a href="/images/news/nix-run-grass.png">
+  <img src="/images/news/nix-run-grass.png" alt="nix run grass" title="Nix running GRASS"
+   width="70%" style="float:center;padding-left:10px;padding-top:20px">
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:57]:
```suggestion
   width="90%" style="float:center;padding-left:10px;padding-top:20px">
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+</a>
+
+To install it, run `nix profile install github:OSGeo/grass#grass`.
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:60]:
```suggestion
If you want to install GRASS instead, run `nix profile install github:OSGeo/grass#grass`.
```

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+Check out [How to use Nix](https://github.com/OSGeo/grass/blob/main/NIX.md) for more instructions.
+
+*Ivan Mincik and Veronica Andreo*
```
Comment (veroandreo) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:64]:
```suggestion
*Ivan Mincik*
```

**Review by veroandreo — CHANGES_REQUESTED** [review]
Thanks @imincik !! Great addition! Let's wait with this news until the doc is merged.

File: content/news/2024_06_07_Nix_environment_and_package_added.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
+---
+title: "Nix development environment and package"
+date: 2024-07-06T12:00:00+02:00
+layout: "news"
+author: Ivan Mincik and the GRASS Dev Team
+---
+
+### You can now develop and run GRASS with Nix
```
Comment (neteler) [inline content/news/2024_06_07_Nix_environment_and_package_added.md:8]:
```suggestion
### You can now develop and run GRASS GIS with Nix
```

Comment (neteler) [reply content/news/2024_06_07_Nix_environment_and_package_added.md:8]: (at least in the section titles...)

**Review by veroandreo — APPROVED** [review]

**Comment by imincik** [comment]
Still doesn't [appear in on the website](https://grass.osgeo.org/news/) . Is it OK ?

**Comment by veroandreo** [comment]
> Still doesn't [appear in on the website](https://grass.osgeo.org/news/) . Is it OK ?

@neteler would you mind checking in the server side? 🙏🏼 It was building just fine locally

**Comment by neteler** [comment]
I gave the cronjob a kick, now there:

https://grass.osgeo.org/news/2024_06_07_nix_environment_and_package_added/

**Comment by veroandreo** [comment]
> I gave the cronjob a kick, now there:
> 
> https://grass.osgeo.org/news/2024_06_07_nix_environment_and_package_added/

Thanks! The code highlighting changes I did were ineffective it seems... This is how it looked like locally:

image

**Comment by veroandreo** [comment]
Yes, indeed! I changed browser and it looks good there! Cool! Thanks! And sorry for the noise :)

**Pull request merged**